### PR TITLE
build: remove cicheck

### LINKED ISSUE #1854: Reconsider use of `redis` build tag
With in the https://github.com/zalando/skipper/pull/1853#discussion_r705322502 it was discovered that `redis` tag condition is not used properly and that redis-dependent tests were always executed.
This is not a bad thing to run as many tests as possible so maybe we should simply eliminate redis build tag altogether.

The idea behind the tag was to exclude tests that depend on the `redis-server` binary. The alternative could be to run all tests by default and introduce a make target to disable redis tests explicitly

### PR DESCRIPTION
**Pull request by AlexanderYastrebov**
Make cicheck is equvalent to shortcheck due to wrong redis build tag use, see https://github.com/zalando/skipper/issues/1854

This change removes redis build tag, build comments and replaces cicheck with the shortcheck target.

Fixes https://github.com/zalando/skipper/issues/1854

### CODE REVIEW

File: delivery.yaml [hunk 1]
```diff
@@ -50,7 +50,7 @@ pipeline:
       fi
       export IMAGE MULTIARCH_IMAGE
 
-      make deps cicheck staticcheck gosec
+      make deps check-fmt vet staticcheck gosec shortcheck
```
Comment (AlexanderYastrebov) [inline delivery.yaml:53]: Move validators before tests to fail early

File: .github/workflows/master.yaml [hunk 2]
```diff
@@ -20,11 +20,9 @@ jobs:
       - run: go version
       - run: make deps
       - run: make check-fmt
-      - run: make build
       - run: make vet
       - run: make staticcheck
       - run: make check-race
-      - run: make cicheck
```
Comment (AlexanderYastrebov) [inline .github/workflows/master.yaml:27]: There is make check-race before

File: .github/workflows/master.yaml [hunk 3]
```diff
@@ -20,11 +20,9 @@ jobs:
       - run: go version
       - run: make deps
       - run: make check-fmt
-      - run: make build
```
Comment (AlexanderYastrebov) [inline .github/workflows/master.yaml:23]: check-race requires build target

File: .github/workflows/master.yaml (hunk 2 above)
Comment (RomanZavodskikh) [reply .github/workflows/master.yaml:27]: So, this step is valid, but redundant, right?

Comment (AlexanderYastrebov) [reply .github/workflows/master.yaml:27]: yes

File: circuit/registry_test.go [hunk 4]
```diff
@@ -1,5 +1,3 @@
-/* +build !race */
```
Comment (AlexanderYastrebov) [inline circuit/registry_test.go:1]: This likely never worked as expected https://pkg.go.dev/go/build#hdr-Build_Constraints

**Comment by RomanZavodskikh** [comment]
:+1:

**Comment by AlexanderYastrebov** [comment]
:+1:

**Pull request merged**